src/gsCore/gsDofMapper.cpp

```diff
@@ -175,6 +180,13 @@ void gsDofMapper::markBoundary( index_t k, const gsMatrix<unsigned> & boundaryDo
     // TO DO: save inverse, eg boundaryDofs(i,0)
 }
 
+void gsDofMapper::markCoupledAsTagged()
+{
+    m_tagged.reserve(m_tagged.size()+m_numCpldDofs);
+    for(int i=0; i< m_numCpldDofs;++i)
+        m_tagged.push_back(m_numFreeDofs-m_numCpldDofs+i);
+}
+
```

Comment:
This function seems buggy: it should insert into m_tagged some indices from m_dofs, not the "cindex" !

Reply:
Depends on what we store. m_tagged contains the index(i,k) of the tagged dofs.

Reply:
Ahhh.. now i see what you mean. So i should store the index i, such that m_dofs[i] is tagged, isnt it? This would make the marking independent on the permutation.

Reply:
wait, but if we only the the indices i, such that m_dofs[i] is tagged, then the calculation of is_tagged_index(..) and tindex(..) is more "complicated".  In order to search in for some gl in m_tagged, we first have to compute its preimage, i.e., the j, such that m_dofs[j] = gl and then do the search based on j in m_tagged...

Reply:
To be consistent, I think you should store the index(i,k) [global index].
ie.
for i=0.. m_dofs[size]
if ( is_coupled_index(m_dof[i])
m_tagged.push_back(m_dof[i]):

sort+remove duplicates before

(essentially you do these stuff in permuteFreeDofs, but it should be done already here..
note that:  m_numFreeDofs-m_numCpldDofs+i has nothing to do with index(i,k)...

Reply:
* Or actually, now that I see it again.. OK now I think your implementation is right here.

Reply:
ok